core/lib/dal/src/consensus/mod.rs

```diff
@@ -100,6 +128,50 @@ impl Payload {
     }
 }
 
+impl ProtoRepr for proto::TransactionV25 {
+    type Type = Transaction;
+
+    fn read(&self) -> anyhow::Result<Self::Type> {
+        use proto::transaction_v25::T;
+        let tx = match required(&self.t)? {
+            T::L1(l1) => abi::Transaction::L1 {
+                tx: required(&l1.rlp)
+                    .and_then(|x| {
+                        let tokens = ethabi::decode(&[abi::L2CanonicalTransaction::schema()], x)
+                            .context("ethabi::decode()")?;
+                        // Unwrap is safe because `ethabi::decode` does the verification.
+                        let tx =
+                            abi::L2CanonicalTransaction::decode(tokens.into_iter().next().unwrap())
+                                .context("L2CanonicalTransaction::decode()")?;
+                        Ok(tx)
+                    })
+                    .context("rlp")?
+                    .into(),
+                factory_deps: l1.factory_deps.clone(),
+                eth_block: 0,
```

Comment:
If `eth_block` still defined if deprecated and unused?

Reply:
eth_block will start to be populated with 0s after https://github.com/matter-labs/zksync-era/pull/2067 is merged and deployed. Until then eth_block is non-zero. Afterwards we will still have non-zero eth_block values for the old certificates (hopefully only for V24 or older). It is not like I can remove it from abi::L1Transaction just like that.